dataactvalidator/scripts/loaderUtils.py

```diff
@@ -71,6 +86,29 @@ def clean_data(data, model, field_map, field_options):
     # trim all columns
     data = data.applymap(lambda x: trim_item(x) if len(str(x).strip()) else None)
 
+    if len(required_values) > 0:
+        # if file is blank, immediately fail
+        if data.empty or len(data.shape) < 2:
+            raise FailureThresholdExceededException(0)
+        # check the columns that must have a valid value, and if they have white space,
+        # replace with NaN so that dropna finds them.
+        for value in required_values:
+            data[value].replace('', np.nan, inplace=True)
+        # drop any rows that are missing required data
+        cleaned = data.dropna(subset=required_values)
+        dropped = data[np.invert(data.index.isin(cleaned.index))]
+        # log every dropped row
+        for index, row in dropped.iterrows():
+            logger.info("Dropped row due to faulty data: fyq:{}--agency:{}--alloc:{}--account:{}".format(
+                row["fiscal_year_quarter"],
+                row['agency_id'], row['allocation_transfer_id'], row['account_number']) +
+                "--pa_code:{}--pa_name:{}".format(row['program_activity_code'], row['program_activity_name']))
+
+        if (len(dropped.index)/len(cleaned.index)) > FAILURE_THRESHOLD_PERCENTAGE:
```

Comment:
question, shouldn't this be `len(dropped.index)/(len(data.index))` if you want it to be an actual percentage?